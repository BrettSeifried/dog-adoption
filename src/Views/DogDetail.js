import DogInfo from '../component/Dog';
import { useEffect, useState } from 'react';
import { deleteDog, getDogsById } from '../services/DogRoute';
import { useParams } from 'react-router-dom';
import { useHistory } from 'react-router-dom';

export default function Home() {
  const { id } = useParams();
  const [info, setInfo] = useState([]);
  const [loading, setLoading] = useState(true);
  const history = useHistory();

  // TRANSLATE TO ASYNC WAIT
  useEffect(() => {
    getDogsById(id)
      .then((data) => setInfo(data))
      .finally(() => setLoading(false));
  }, [id]);

  const handleDelete = async (e) => {
    try {
      await deleteDog(id);
      alert('You have deleted a dog.');
    } catch {
      alert('Delete failed');
    }
    e.preventDefault();
    history.push('/');
  };

  if (loading) return <h1> Welcome for considering helping a lost dog </h1>;

  return (
    <ul>
      <DogInfo {...info} handleDelete={handleDelete} />
    </ul>
  );
}
