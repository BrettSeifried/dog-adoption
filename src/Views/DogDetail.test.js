import { render, screen } from '@testing-library/react';
import { Route, MemoryRouter } from 'react-router-dom';
import DogDetail from './DogDetail';

test('Should render individual Dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/2']}>
      <Route exact path="/dogs/:id" component={DogDetail} />
    </MemoryRouter>
  );
  await screen.findByText('Cato');

  expect(container).toMatchSnapshot();
});
