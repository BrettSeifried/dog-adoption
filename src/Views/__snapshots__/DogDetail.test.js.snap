// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render individual Dog 1`] = `
<div>
  <ul>
    <div
      class="dog-info"
    >
      <section
        class="dog-text"
      >
        <h1>
          Cato
        </h1>
        <img
          alt="dog"
          src="https://placedog.net/1000/563?id=199"
        />
        <h3>
          6
           Year old 
          Terrier
        </h3>
        <p>
          Loves to eat all poo poo
        </p>
        <button>
          <a
            href="/dogs/2/edit"
          >
             Edit 
          </a>
        </button>
        <button>
          Delete
        </button>
      </section>
    </div>
  </ul>
</div>
`;
